src/strongbox/gitlab_api.clj

```diff
@@ -0,0 +1,141 @@
+         :assets
+         :links
+         (filter (juxt false? :external))
+         (filter (juxt supported-link-types :link_type))
+         (mapcat link-release)
+         (remove nil?))))
+
+(defn-spec expand-summary (s/or :ok :addon/release-list, :error nil?)
+  [addon-summary :addon/expandable, game-track ::sp/game-track]
+  ;; todo: sink errors
+  (let [result (-> addon-summary :source-id api-url (str "/releases") http/download-with-backoff utils/from-json)
+        known-game-tracks (-> addon-summary :game-track-list utils/nilable (or [:retail]))
+        release-list (->> result
+                          (remove :upcoming_release)
+                          (mapcat #(parse-release % known-game-tracks))
+                          (group-by :game-track))]
+    (get release-list game-track)))
+
+;; ---
+
+(defn-spec base64-decode (s/or :ok? string?, :error nil?)
+  [string string?]
+  (String. (.decode (Base64/getDecoder) string)))
+
+(defn-spec download-decode-blob map?
+  [url ::sp/url]
+  (->> url http/download utils/from-json :content base64-decode toc/-parse-toc-file))
+
+(defn-spec find-toc-files map?
+  "returns a map of {toc-filename->blob-url}"
+  [source-id :addon/source-id]
+  (let [result (-> source-id
+                   api-url
+                   (str "/repository/tree")
+                   http/download
+                   utils/from-json)
+
+        blob-url (fn [item]
+                   ;; {"Foo.toc" "https://gitlab.com/api/v4/projects/foo%2Fbar/repository/blobs/125c899d813d2e11c976879f28dccc2a36fd207b"}
+                   {(:name item) (str (api-url source-id) "/repository/blobs/" (:id item))})
+
+        toc-files (->> result
+                       (filter #(-> % :path lower-case (ends-with? ".toc")))
+                       (map blob-url)
+                       (into {}))]
+
+    toc-files))
+
+(defn-spec guess-game-track-list ::sp/game-track-list
+  [source-id :addon/source-id]
+  (let [toc-file-map (find-toc-files source-id)]
+    ;; if we have multiple toc files, assume multi-toc and check for prefixes
+    (if (> (count toc-file-map) 1)
+      (let [key-check (fn [key]
+                        (cond
+                          (ends-with? key "-bcc.toc") :classic-tbc
+                          (ends-with? key "-classic.toc") :classic
+                          :else :retail))]
+        (->> toc-file-map keys (map lower-case) (map key-check) set sort vec))
+
+      ;; otherwise, download toc file and analyse it's contents
+      ;; todo: test, no toc-file-list is empty
+      (->> toc-file-map
+           vals ;; blob urls
+           first
+           download-decode-blob
+           (select-keys* [:interface :#interface])
+           vals
+           (map utils/to-int)
+           (map utils/interface-version-to-game-track)
+           (remove nil?)
+           set
+           vec))))
+
+(defn-spec parse-user-string (s/or :ok :addon/source-id :error nil?)
+  "extracts the addon ID from the given `url`."
```

Comment:
I doubled down on the string version of the IDs - remove/increase the max width of the GUI field